whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileProcessor.java

```diff
@@ -0,0 +1,69 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.dao.NrtmSource;
+import net.ripe.db.nrtm4.dao.NrtmSourceHolder;
+import net.ripe.db.nrtm4.dao.SnapshotFile;
+import net.ripe.db.nrtm4.domain.PublishableSnapshotFile;
+import net.ripe.db.nrtm4.jmx.NrtmProcessControl;
+import org.mariadb.jdbc.internal.logging.Logger;
+import org.mariadb.jdbc.internal.logging.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Optional;
+
+
+@Service
+public class NrtmFileProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NrtmFileProcessor.class);
+
+    private final NrtmFileService nrtmFileService;
+    private final NrtmProcessControl nrtmProcessControl;
+    private final NrtmSourceHolder nrtmSourceHolder;
+    private final SnapshotFileGenerator snapshotFileGenerator;
+
+    public NrtmFileProcessor(
+        final NrtmFileService nrtmFileService,
+        final NrtmProcessControl nrtmProcessControl,
+        final NrtmSourceHolder nrtmSourceHolder,
+        final SnapshotFileGenerator snapshotFileGenerator
+    ) {
+        this.nrtmFileService = nrtmFileService;
+        this.nrtmProcessControl = nrtmProcessControl;
+        this.nrtmSourceHolder = nrtmSourceHolder;
+        this.snapshotFileGenerator = snapshotFileGenerator;
+    }
+
+    public void updateNrtmFilesAndPublishNotification() {
+        LOGGER.info("runWrite() called");
+        final NrtmSource source = nrtmSourceHolder.getSource();
+        final Optional<SnapshotFile> lastSnapshot = snapshotFileGenerator.getLastSnapshot(source);
+        Optional<PublishableSnapshotFile> publishableSnapshotFile = Optional.empty();
+        if (lastSnapshot.isEmpty()) {
+            LOGGER.info("No previous snapshot found");
+            if (nrtmProcessControl.isInitialSnapshotGenerationEnabled()) {
+                LOGGER.info("Initializing...");
```

Comment:
Let's discuss how we can separate the logic of creating initial snapshot from the main code. As it will be required only once having a if condition here is less readable

Reply:
leaving as-is for now -- added a note in the NRTM todo list for review later.